src/components/ColorPicker/components/ColorPickerItemComponent/ColorPickerItemComponent.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useRef, useCallback, useMemo, forwardRef } from "react";
+import cx from "classnames";
+import {
+  COLOR_STYLES,
+  COLOR_STYLES_VALUES,
+  CONTENT_COLORS_VALUES,
+  contentColors
+} from "../../../../utils/colors-vars-map";
+import ColorUtils from "../../../../utils/colors-utils";
+import "./ColorPickerItemComponent.scss";
+import Icon from "../../../Icon/Icon";
+import Tooltip from "../../../Tooltip/Tooltip";
+import Clickable from "../../../Clickable/Clickable";
+import { COLOR_SHAPES, COLOR_SHAPES_VALUES } from "../../ColorPickerConstants";
+import { getTestId } from "../../../../tests/test-ids-utils";
+import { ComponentDefaultTestId } from "../../../../tests/constants";
+import { SubIcon, VibeComponent, VibeComponentProps, ElementContent } from "../../../../types";
+import { BASE_SIZES_VALUES } from "../../../../constants";
+
+export interface ColorPickerItemComponentProps extends VibeComponentProps {
+  color: CONTENT_COLORS_VALUES | string;
```

Comment:
In case of internal components which are building blocks of our design system exported components, we are not really require to use the VibeComponentProps and VibeComponent types. 
The purpose of these types is to make sure that all our exported components contain the mandatory props (class name, data test id, id). 
By the way, right now these props are not mandatory for some reason (which is bad, not sure why we choose to do this), maybe worth to convert it.
---
In internal props we are not really need to keep the same props and structure as in the exported props so there is no need to use these types (for example this component not support class name prop which should be supported in all external components)

Reply:
I actually think it's a good practice to have the same props convention all across components, even if they are internal.

For example -`className` and `data-testid` are both really important on all levels.
So maybe the `VibeComponent`'s `ref` forwarding isn't necessary, but I'd say that the common props should be always defined